### ISSUE TITLE
When comparing, limit to facets in comparison

### ISSUE THREAD

**Issue by wanderingstan** [error-central/diffenv#14]
Occurs to me that when when coworker shares a "diffenv" with me (and asks for me to compare with my env), we should by default run _only_ the facets that were included in the original. This (1) helps with security and (2) allows the sharer to focus on the facets they're most interested in. E.g. if they didn't include the `php` facet, then neither should I, regardless of what's in my config. 

A fancier version of the `--share` option could even pop open an editor (as in a git commit) to allow the sender to prever/edit their diffenv before sharing it. 

@werg , thoughts?

**Comment by werg** [comment]
Yes, I'd agree. If there's a repo config, should it take precedence?

**Comment by wanderingstan** [comment]
My gut says that when doing a `--compare`, that the incoming env sets determines what we're comparing.

I suspect config file usage will be for power users, and if it was overriding passed-in files it would be quite confusing.

Comparing my env to yours will be the #1 use case, and there it makes the most sense that if I send you an env with my `os`,`python`, and `git` facets, then those are exactly the facets that I want to see compared with yours. 

In this light, the config for a repo should be seen as the "suggested list of facets most relevant to this project", but once an env file has been generated, then it takes precedence.

But then, this is all just guessing. We need to get to dog-fooding this asap. :)
